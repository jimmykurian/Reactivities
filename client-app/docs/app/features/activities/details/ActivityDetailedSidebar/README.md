[**Client App Documentation v0.0.0**](../../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../../README.md) / app/features/activities/details/ActivityDetailedSidebar

# app/features/activities/details/ActivityDetailedSidebar

## Index

### Functions

- [default](functions/default.md)
