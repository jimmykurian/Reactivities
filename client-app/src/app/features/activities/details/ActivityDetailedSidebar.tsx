/**
 * @file ActivityDetailedSidebar.tsx
 * @author Jimmy Kurian
 * @fileoverview This file contains the ActivityDetailedSidebar component.
 */

import { Segment, List, Label, Item, Image } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { observer } from 'mobx-react-lite';

/**
 * @component ActivityDetailedSidebar
 * @description The ActivityDetailedSidebar component displays a sidebar with a list of people going to the activity.
 *
 * @returns {JSX.Element} The rendered ActivityDetailedSidebar component.
 *
 * @remarks
 * This component uses Semantic UI React components to create a styled sidebar. It includes a segment header displaying the number of people going to the activity and a list of attendees. Each attendee is shown with an image, name, and an optional "Following" label. The host is marked with an orange ribbon.
 *
 * @example
 * Here is an example of how to use the ActivityDetailedSidebar component:
 * ```tsx
 * import React from 'react';
 * import ActivityDetailedSidebar from './features/activities/details/ActivityDetailedSidebar';
 *
 * const App = () => (
 *   <div>
 *     <ActivityDetailedSidebar />
 *   </div>
 * );
 *
 * export default App;
 * ```
 */
export default observer(function ActivityDetailedSidebar(): JSX.Element {
  return (
    <>
      <Segment
        textAlign="center"
        style={{ border: 'none' }}
        attached="top"
        secondary
        inverted
        color="teal"
      >
        3 People Going
      </Segment>
      <Segment attached>
        <List relaxed divided>
          <Item style={{ position: 'relative' }}>
            <Label
              style={{ position: 'absolute' }}
              color="orange"
              ribbon="right"
            >
              Host
            </Label>
            <Image size="tiny" src={'/assets/user.png'} />
            <Item.Content verticalAlign="middle">
              <Item.Header as="h3">
                <Link to={`#`}>Bob</Link>
              </Item.Header>
              <Item.Extra style={{ color: 'orange' }}>Following</Item.Extra>
            </Item.Content>
          </Item>

          <Item style={{ position: 'relative' }}>
            <Image size="tiny" src={'/assets/user.png'} />
            <Item.Content verticalAlign="middle">
              <Item.Header as="h3">
                <Link to={`#`}>Tom</Link>
              </Item.Header>
              <Item.Extra style={{ color: 'orange' }}>Following</Item.Extra>
            </Item.Content>
          </Item>

          <Item style={{ position: 'relative' }}>
            <Image size="tiny" src={'/assets/user.png'} />
            <Item.Content verticalAlign="middle">
              <Item.Header as="h3">
                <Link to={`#`}>Sally</Link>
              </Item.Header>
            </Item.Content>
          </Item>
        </List>
      </Segment>
    </>
  );
});
