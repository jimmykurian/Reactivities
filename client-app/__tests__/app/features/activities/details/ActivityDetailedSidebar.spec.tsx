import { render, screen } from '@testing-library/react';
import ActivityDetailedSidebar from '../../../../../src/app/features/activities/details/ActivityDetailedSidebar';
import { MemoryRouter } from 'react-router-dom';

// Mocking observer from mobx-react-lite
jest.mock('mobx-react-lite', () => ({
  observer: (component: React.ComponentType) => component,
}));

describe('ActivityDetailedSidebar', () => {
  const renderComponent = () => {
    render(
      <MemoryRouter>
        <ActivityDetailedSidebar />
      </MemoryRouter>,
    );
  };

  test('renders ActivityDetailedSidebar component', () => {
    // Act
    renderComponent();

    // Assert
    expect(screen.getByText('3 People Going')).toBeInTheDocument();
    expect(screen.getByText('Bob')).toBeInTheDocument();
    expect(screen.getByText('Tom')).toBeInTheDocument();
    expect(screen.getByText('Sally')).toBeInTheDocument();
    expect(screen.getAllByText('Following')).toHaveLength(2);
    expect(screen.getAllByRole('img')).toHaveLength(3);
  });

  test('matches snapshot', () => {
    // Arrange & Act
    const { asFragment } = render(
      <MemoryRouter>
        <ActivityDetailedSidebar />
      </MemoryRouter>,
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
