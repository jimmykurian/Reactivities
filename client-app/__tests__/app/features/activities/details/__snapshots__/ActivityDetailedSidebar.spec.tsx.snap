// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityDetailedSidebar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui teal inverted secondary top attached center aligned segment"
  >
    3 People Going
  </div>
  <div
    class="ui attached segment"
  >
    <div
      class="ui divided relaxed list"
      role="list"
    >
      <div
        class="item"
        style="position: relative;"
      >
        <div
          class="ui orange right ribbon label"
          style="position: absolute;"
        >
          Host
        </div>
        <img
          class="ui tiny image"
          src="/assets/user.png"
        />
        <div
          class="middle aligned content"
        >
          <h3
            class="header"
          >
            <a
              href="/"
            >
              Bob
            </a>
          </h3>
          <div
            class="extra"
            style="color: orange;"
          >
            Following
          </div>
        </div>
      </div>
      <div
        class="item"
        style="position: relative;"
      >
        <img
          class="ui tiny image"
          src="/assets/user.png"
        />
        <div
          class="middle aligned content"
        >
          <h3
            class="header"
          >
            <a
              href="/"
            >
              Tom
            </a>
          </h3>
          <div
            class="extra"
            style="color: orange;"
          >
            Following
          </div>
        </div>
      </div>
      <div
        class="item"
        style="position: relative;"
      >
        <img
          class="ui tiny image"
          src="/assets/user.png"
        />
        <div
          class="middle aligned content"
        >
          <h3
            class="header"
          >
            <a
              href="/"
            >
              Sally
            </a>
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
